### DISCUSSION TITLE
401 Unauthorized response from vercel

### DISCUSSION THREAD

**Discussion by Kanezal** [vercel/community#5519] [Bug Report] [answered]
### Page to Investigate

https://discord-express-bot.vercel.app/api/interactions

### Steps to Reproduce

POST to https://discord-express-bot.vercel.app/api/interactions with 0 params
![image](https://github.com/vercel/community/assets/58330966/6a376bf1-23d2-4068-9816-044c4b1c7369)
*401 Unauthorized!!!* But you also getting correct response from discord interactions, but I can't put my vercel project in interaction endpoint in my discord bot, because response status is 401.

Let's take a look at the settings of the vercel project:
![image](https://github.com/vercel/community/assets/58330966/a7ecd464-a1f6-40ab-8585-0579181f2201)
*Nothing!*

**Bug confirmed. Please fix.**

P.s. Problem arrives after 14.01.2024

> If you think it is a bug in my source code, it isn't true. I checked it using ngrok and endpoint url saved correctly.

**Comment by amyegan** [comment]
I don't think this error is coming from Vercel directly. It may be an issue with an attempted connection to Discord or another service. Some of the first things I would check are the environment variables related to that connection. I'd also recommend looking at the [Runtime Logs](https://vercel.com/docs/observability/runtime-logs#) to get more detailed info about what went wrong.

**Reply by Kanezal** [reply]
Thanks for the comment, however, I would ask you to verify the validity of the response from Vercel using Postman, which in turn would eliminate the possibility of problems with Discord. At the same time, Postman also indicates the 401 status. There is no useful information in the execution logs, except that the 401 status is indicated in the response. 
The 401 status problem disappears when running the code locally and using **ngrok**, which indicates an error in Vercel. Also, this error appeared after a certain date, while the code **_has not changed_** for a long time.

**Reply by techrook** [reply]
i have the same issue too

**Comment by anshroy2** [comment]
I'm also facing a similar problem. I have my vercel set up with python and a post request, but if I try to post any data, it returns a 401 unauthorized. This is not even appearing in the logs of vercel...

Let me know if more information is needed.

**Comment by amyegan** [comment, marked as answer]
I'm trying to determine where the error is coming from, but I don't have enough information. Can you share a screenshot of the error? A [minimal reproducible example](https://vercel.com/guides/creating-a-minimal-reproducible-example) would also be helpful.

It could be that [Deployment Protection](https://vercel.com/docs/security/deployment-protection) is enabled and blocking access. Another possibility is a missing or invalid environment variable which would result in a 401 error being passed through from a third-party service used in the app.

**Reply by eriese** [reply]
for me it was deployment protection. Thanks!

**Reply by techrook** [reply]
same here.

[a comment by an automated account is not shown]
